pkg/standalone/publish.go

```diff
@@ -43,19 +42,22 @@ func SendPayloadToTopic(topic, payload, pubsubName string) error {
 	url := fmt.Sprintf("http://localhost:%s/v%s/publish/%s/%s", fmt.Sprintf("%v", daprHTTPPort), api.RuntimeAPIVersion, pubsubName, topic)
 	// nolint: gosec
 	r, err := http.Post(url, "application/json", bytes.NewBuffer(b))
+	if err != nil {
+		return err
+	}
 
-	if r != nil {
-		defer r.Body.Close()
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("unexpected status code %d on publishing to %s in %s", r.StatusCode, topic, pubsubName)
 	}
```

Comment:
May be move this to inside the `r != nil` block to avoid dereferencing `nil` pointer, since you already have a `nil` check block down at line 53?

However... from the doc of `http` package I think `err == nil => r != nil` (see examples here https://golang.org/pkg/net/http ). So you probably can (1) remove the `nil` check altogether, (2) move the `defer r.Body.Close()` to before the status code check, so we close the body even not 200.

Reply:
done